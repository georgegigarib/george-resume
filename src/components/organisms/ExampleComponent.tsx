import { useTranslation } from 'react-i18next'

const ExampleComponent = () => {
  const { t } = useTranslation()
  return (
    <div className="p-4 text-blue-900 rounded flex justify-center">
      <h4>{t('underConstruction')}</h4>
    </div>
  )
}

export default ExampleComponent
