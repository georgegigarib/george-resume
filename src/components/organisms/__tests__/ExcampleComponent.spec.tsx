import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import ExampleComponent from '@/components/organisms/ExampleComponent'

describe('ExampleComponent', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<ExampleComponent />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the component with a random class', () => {
    render(<ExampleComponent />)

    const container = screen.getByTestId('example-container')
    const classRegex = /bg-app-lightModeBoxes-box\d+|dark:bg-app-darkModeBoxes-box\d+/
    expect(container.className).toMatch(classRegex)
  })

  it('renders the image correctly', () => {
    render(<ExampleComponent />)

    const image = screen.getByRole('img')
    expect(image).toBeDefined()
    expect(image).toHaveClass('rounded-lg min-h-full min-w-full flex-shrink object-cover')
  })

  it('generates a different random class on re-render', () => {
    const { unmount } = render(<ExampleComponent />)

    const firstContainer = screen.getByTestId('example-container')
    const firstClassName = firstContainer.className

    unmount()

    render(<ExampleComponent />)

    const secondContainer = screen.getByTestId('example-container')
    const secondClassName = secondContainer.className

    expect(firstClassName).not.toBe(secondClassName)
  })
})
