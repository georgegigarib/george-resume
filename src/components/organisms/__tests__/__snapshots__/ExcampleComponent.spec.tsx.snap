// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExampleComponent > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-blue-900 rounded-xl flex justify-center flex-wrap items-center contain-size overflow-hidden h-full border-[2px] bg-app-lightModeBoxes-box14 dark:bg-app-darkModeBoxes-box14 border-app-darkModeBoxes-box14 dark:border-app-lightModeBoxes-box14"
    data-testid="example-container"
  >
    <div
      class="absolute inset-0 flex items-center justify-center"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <img
      class="rounded-lg min-h-full min-w-full flex-shrink object-cover opacity-0"
      src="/src/assets/images/construction.png"
    />
  </div>
</DocumentFragment>
`;
